src/test/cc/math/distributed_discrete_gaussian_noiser_test.cc

```diff
@@ -0,0 +1,127 @@
+      {.contributor_count = 1,
+       .sigma = 6.87,
+       .truncate_threshold = 13,
+       .shift_offset = 13});
+
+  ASSERT_THAT(distributed_discrete_gaussian_noiser.GenerateNoiseComponent(),
+              IsOk());
+}
+
+TEST(DiscreteGaussianNoiserGenerateNoiseComponent, ReturnsSampledValues) {
+  DistributedDiscreteGaussianNoiser distributed_discrete_gaussian_noiser(
+      {.contributor_count = 1,
+       .sigma = 6.87,
+       .truncate_threshold = 13,
+       .shift_offset = 13});
+  int num = 5;
+  std::vector<int64_t> results;
+
+  for (size_t i = 0; i < num; ++i) {
+    ASSERT_OK_AND_ASSIGN(
+        int64_t temp,
+        distributed_discrete_gaussian_noiser.GenerateNoiseComponent());
+    results.push_back(temp);
+  }
+
+  ASSERT_THAT(results.size(), num);
+}
+
+TEST(DiscreteGaussianNoiserIndividualComponent, MinMaxShouldBeCorrect) {
+  double sigma = 48.231;
+  int64_t offset = 261;
+
+  int64_t min_value = 1000;
+  int64_t max_value = 0;
+
+  size_t num_trials = 10000;
+
+  DistributedDiscreteGaussianNoiser distributed_gaussian_noiser(
+      {.contributor_count = 1,
+       .sigma = sigma,
+       .truncate_threshold = offset,
+       .shift_offset = offset});
+
+  for (size_t i = 0; i < num_trials; ++i) {
+    ASSERT_OK_AND_ASSIGN(int64_t temp,
+                         distributed_gaussian_noiser.GenerateNoiseComponent());
+
+    min_value = std::min(min_value, temp);
+    max_value = std::max(max_value, temp);
+  }
+
+  // Max should be smaller than 2 * shift_offset.
+  ASSERT_LE(max_value, 2 * offset);
+  // Min should be larger than 0.
+  ASSERT_GE(min_value, 0);
+}
+
+TEST(DiscreteGaussianNoiserGlobalSummation,
+     ProbabilityMassFunctionShouldBeCorrect) {
+  int64_t contributor_count = 1;  // 1 contributor
+  double sigma = 48.231;
+  int64_t offset = 261;
+
+  DistributedDiscreteGaussianNoiseComponentOptions options = {
+      .contributor_count = contributor_count,
+      .sigma = sigma,
+      .truncate_threshold = offset,
+      .shift_offset = offset};
+  DistributedDiscreteGaussianNoiser distributed_discrete_gaussian_noiser(
+      options);
+
+  int64_t min_output = 0;
+  int64_t max_output = 2 * offset;
+
+  size_t num_trials = 10000;
+  std::map<int64_t, size_t> frequency_distribution;
+
+  for (size_t i = 0; i < num_trials; ++i) {
+    ASSERT_OK_AND_ASSIGN(
+        int64_t temp,
+        distributed_discrete_gaussian_noiser.GenerateNoiseComponent());
+    ASSERT_GE(temp, min_output);
+    ASSERT_LE(temp, max_output);
+    ++frequency_distribution[temp];
+  }
+
+  for (auto const& x : frequency_distribution) {
+    double probability = static_cast<double>(x.second) / num_trials;
+    // Expected probability is Gaussian(normal) distribution.
+    double expected_probability =
+        std::exp(-std::pow((x.first - offset), 2) / (2 * std::pow(sigma, 2))) /
+        (sigma * std::sqrt(2 * M_PI));
+
+    EXPECT_NEAR(probability, expected_probability, 0.01);
+  }
+}
+
```

Comment:
- I still think we should add tests for `contributor_count` > 1 here because this is about making sure the sampling itself is correct and not just the `offset` computation which was tested in `src/test/cc/math/noise_parameters_computation_test.cc`.
- I also think that perhaps we should add tests for varying values of `offset`, `truncate_threshold` and `sigma` here just to be on the safe side.

Wdyt?